src/main/java/com/google/devtools/build/lib/remote/RemoteCache.java

```diff
@@ -323,6 +325,32 @@ public void download(
       FileOutErr origOutErr,
       OutputFilesLocker outputFilesLocker)
       throws ExecException, IOException, InterruptedException {
+    if (actionResultDownloader != null) {
```

Comment:
Can you explain the purpose of that? I would think that eager downloading should not really interact with a lazy FUSE/NFS solution -- we should install all outputs there always and it's up to the user what they will choose to access (which would trigger a download).

Reply:
> Can you explain the purpose of that? I would think that eager downloading should not really interact with a lazy FUSE/NFS solution -- we should install all outputs there always and it's up to the user what they will choose to access (which would trigger a download).

This is exactly what bb\_clientd does. So even though the thing is called `ActionResultDownloader` it doesn't actually trigger a download. Maybe a better name would be `ActionResultInstantiator` or something?